sender.go

```diff
@@ -159,6 +159,11 @@ func (s *Sender) Send(msg *Message, retries int) (*Response, error) {
 // updateStatus updates the status of the messages sent to devices and
 // returns the number of recoverable errors that could be retried.
 func updateStatus(msg *Message, resp *Response, allResults map[string]Result) int {
+	if msg.To != "" {
+		if resp.Results[0].Error != "Unavailable" {
```

Comment:
Does FCM ever return "Unavailable" on a failed topic send? And if so, shouldn't the function return a non-zero value? It looks like it'll fall back to the length of unsentRegIDs which will be zero.

Reply:
I copied "Unavailable" from a few lines down. The existing code treats this as the only retryable error. You're right about the non-zero return.